### ISSUE TITLE
Use Github releases instead of putting binaries in the git repository.

### ISSUE THREAD

**Issue by xorgy** [KhronosGroup/glslang#48]
Git is very fast and efficient for tracking line-oriented text files, however it is fairly weak at working with binaries. The common case is that git will rewrite the entire binary file, and both will need to be downloaded with the repository, making Git clones(and some other operations) slower than they need to be. It will get far slower over time if binaries are updated in the repository more and more times.

Github has [a mechanism for releasing binaries](https://help.github.com/articles/creating-releases/). It's pretty convenient and makes it simple for people to pick up older releases if the need may arise.

**Comment by johnkslang** [comment]
Understood, and I don't intend to use git commits to track executables.  The current intention is to keep the longstanding Khronos location (see link at the beginning of the Readme) to hold the binaries.  This should be where folks wanting a binary are steered.  The recent git commit for executables was an exception, and I might rather delete that part of the repository, if that's possible.

Up to date binaries are now at Khronos.

**Comment by jekstrand** [comment]
I would very much like to see the binaries pulled out.  I spent almost an entire day a week ago trying to figure out why Install/Linux/glslangValidator was stuck at rev. 30 even though I was rebuilding against newer sources.  Turns out, my build was somewhere else entirely and Install/Linux/glslangValidator was a red herring.

**Comment by johnkslang** [comment]
@jekstrand, agreed.  Commit ebb7bf37fd4681dd33e6dabdb3401187f7eef59f removes them.
@xorgy, leaving issue open for future use of GitHub Releases.  (They are currently hosted by Khronos.)

**Comment by xorgy** [comment]
Thank you, and welcome back. :-)

**Comment by clshortfuse** [comment]
Will releases get binaries under assets? I'm not entirely sure why main-tot has binaries but numbered releases don't. It would very useful to have pinned versions instead of an inconsistent `main-tot` reference that goes out of sync every commit. Without pinned releases, I can see a lot of issues where one person downloaded a main-tot binary that much older than another team member.

For reference, Microsoft uses releases with binaries such as https://github.com/microsoft/DirectXShaderCompiler/releases/download/v1.9.2607/dxc_2026_07_29.zip

Slang as well such as https://github.com/shader-slang/slang/releases/download/v2026.14.1/slang-2026.14.1-windows-x86_64.zip

I can't seem to find any official Khronos binaries for releases but did find LunarG's Vulkan SDK seems to package it, but no idea what version it used and a ~250mb exe installer for one binary is overkill.

I guess I could fork the repo, build my own cmake that takes the main-tot workflow, though seems odd.

**Comment by clshortfuse** [comment]
lol https://github.com/KhronosGroup/glslang/commit/04a35a9f4f69bb5593e9339f1838a06640316c6d

5 minutes ago

**Comment by jeremy-lunarg** [comment]
I'm building a set of release archives for 16.4.0 as we speak. Should land soon.

https://github.com/KhronosGroup/glslang/actions/runs/30645661303

**Comment by jeremy-lunarg** [comment]
That didn't work. Let me see what happened.

**Comment by jeremy-lunarg** [comment]
Oh wait. Duh! I can't apply this to our existing 16.4.0 release because my changes don't exist in that commit.

**Comment by jeremy-lunarg** [comment]
@clshortfuse We've released [16.5.0](https://github.com/KhronosGroup/glslang/releases/tag/16.5.0). It has release archives.
